decidim-proposals/lib/decidim/api/proposal_type.rb

```diff
@@ -66,6 +97,16 @@ def self.authorized?(object, context)
       rescue Decidim::PermissionAction::PermissionNotSetError
         false
       end
+
+      private
+
+      def current_settings
+        object.component.current_settings
+      end
+
+      def proposal_has_costs?
+        object.cost.present?
```

Comment:
Should also check the `published_state?` method, if not we will leak this information